server/models/referral.model.js

```diff
@@ -62,6 +62,25 @@ const ReferralSchema = new mongoose.Schema(
         type: Date,
       },
     },
+    assignedReferral: {
+      total: {
+        type: Number,
+        default: 0,
+      },
+      transactions: [
+        {
+          transactionId: {
+            type: ObjectId,
+          },
+          percentage: {
+            type: Number,
+          },
+          amountPayable: {
```

Comment:
Changing this for consistency with the amount in reward
```suggestion
          amount: {
```